src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java

```diff
@@ -277,4 +278,43 @@ public static long fileSize(String path, PipelineOptions popts) throws IOExcepti
             return new File(path).length();
         }
     }
+
+    /**
+     * Returns the total file size of all files in a directory, or the file size if the path specifies a file.
+     * Note that sub-directories are ignored - they are not recursed into.
+     * Only supports HDFS and local paths.
+     *
+     * @param path The URL to the file or directory whose size to return
+     * @param popts PipelineOptions for GCS (if relevant; otherwise pass null)
+     * @return the total size of all files in bytes
+     */
+    public static long dirSize(String path, PipelineOptions popts) {
+        // GCS case
+        if (isCloudStorageUrl(path)) {
+            // TODO: implement directory listing
+            throw new UnsupportedOperationException("Directory size listing not supported on GCS.");
+        }
+        // local file or HDFS case
+        try {
+            Path hadoopPath = new Path(path);
+            FileSystem fs = new Path(path).getFileSystem(new Configuration());
+            FileStatus status = fs.getFileStatus(hadoopPath);
+            if (status == null) {
+                throw new UserException.CouldNotReadInputFile(path, "File not found.");
+            }
+            long size = 0;
+            if (status.isDirectory()) {
+                for (FileStatus st : fs.listStatus(status.getPath())) {
+                    if (!st.isDirectory()) {
+                        size += st.getLen();
```

Comment:
Should you use `st.isFile()` here instead to account for the possibility of symlinks in the directory?